src/sysusers/sysusers.c

```diff
@@ -924,6 +991,12 @@ static int write_files(Context *c) {
 
         assert(c);
 
+        /* Immediately mark users and groups as not failed, iff there is nothing to write. */
+        if (ordered_hashmap_isempty(c->todo_uids))
+                *ret_users_failed = false;
+        if (ordered_hashmap_isempty(c->todo_gids) && ordered_hashmap_isempty(c->members))
+                *ret_groups_failed = false;
```

Comment:
our naming rules dictate that ret_xyz is how return params that are filled in on success should be named. but these ones are not like that, they are filled in even if the function fails ultimately.